### DISCUSSION TITLE
how to play video from url?

### DISCUSSION THREAD

**Discussion by zoreu** [PaulleDemon/tkVideoPlayer#17] [Q&A] [answered]
how to play video from url?

**Comment by PaulleDemon** [comment, marked as answer]
Hello @zoreu , if you are looking to play youtube videos, then install `pip install pafy` and `pip install youtube-dl==2020.12.2` <- install this specific version. Now check out the following code:

```py
import pafy

import tkinter as tk
from tkVideoPlayer import TkinterVideo

url = "" #youtube link

video = pafy.new(url)
best = video.getbest(preftype="mp4")

root = tk.Tk()

videoplayer = TkinterVideo(master=root, scaled=True)
videoplayer.load(best.url)
videoplayer.pack(expand=True, fill="both")

videoplayer.play() # play the video

root.mainloop()
```

If you have any further questions let me know, otherwise mark this as answer.
